src/StyleProvider.js

```diff
@@ -36,6 +28,16 @@ export default class StyleProvider extends PureComponent {
     };
   }
 
+  componentDidUpdate(prevProps) {
+    const { theme } = this.state;
```

Comment:
Elsewhere you destructure props first. Can we always do props first then state? Make a convention out of it?
E.g.:
1. Props `const { derp } = this.props;`
2. State `const { herp } = this.state;
3. Arguments `const { lerp } = methodArgument;`